libraries/AP_HAL_Linux/SPIUARTDriver.cpp

```diff
@@ -131,49 +108,15 @@ int SPIUARTDriver::_write_fd(const uint8_t *buf, uint16_t size)
      * this operation since it's the same as in the
      * UARTDriver::write().
      */
-
-    uint8_t *buffer = _buffer;
-
-    uint16_t _head, space;
-    space = BUF_SPACE(_readbuf);
-
-    if (space == 0) {
-        return ret;
-    }
-
-    if (size > space) {
-        size = space;
-    }
-
-    if (_readbuf_tail < _head) {
-        // perform as single memcpy
-        assert(_readbuf_tail+size <= _readbuf_size);
-        memcpy(&_readbuf[_readbuf_tail], buffer, size);
-        BUF_ADVANCETAIL(_readbuf, size);
-        return ret;
-    }
-
-    // perform as two memcpy calls
-    uint16_t n = _readbuf_size - _readbuf_tail;
-    if (n > size) n = size;
-    assert(_readbuf_tail+n <= _readbuf_size);
-    memcpy(&_readbuf[_readbuf_tail], buffer, n);
-    BUF_ADVANCETAIL(_readbuf, n);
-    buffer += n;
-    n = size - n;
-    if (n > 0) {
-        assert(_readbuf_tail+n <= _readbuf_size);
-        memcpy(&_readbuf[_readbuf_tail], buffer, n);
-        BUF_ADVANCETAIL(_readbuf, n);
-    }
+    _readbuf.write(_buffer, size);
 
     return ret;
 }
 
-static uint8_t ff_stub[300] = {0xff};
-
 int SPIUARTDriver::_read_fd(uint8_t *buf, uint16_t n)
 {
+    static uint8_t ff_stub[100] = {0xff};
```

Comment:
static variables are not allowed, please make a member function. if for some reason this class gets reused in the future, a static variable would cause incorrect behavior. Using a member variable takes up the same amount of space (in single class case) and is safer.